### PR TITLE
fix: Spaces are not allowed on widgets children names (tabs, menus, buttons etc.)

### LINKED ISSUE #9443: [Bug]: Spaces are not allowed on widgets children names (tabs, menus, buttons etc.)
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

User cannot use space in tab widget naming. Space can only be included as an afterthought.

[LOOM DEMO](https://www.loom.com/share/9485d1bd690b4c9f99855fffd57fdd45)

### Steps To Reproduce

1. Go to tab widget and try to name it First Tab. Since space isn't allowed, type FirstTab and then try to add in a space after typing out the entire name and observe that this is possible.

### Environment

Release

### Version

Cloud

### PR DESCRIPTION
**Pull request by wmdev0808**
## Description
- User cannot use space in tab widget naming. Space can only be included as an afterthought.
- This issue is due to recent code changes to the relevant property controls

Fixes #9443 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Test A
- Test B

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: fix/9443-spaces-not-allowed-on-widget-children-names 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 55.44 **(0.01)** | 37 **(0)** | 35.3 **(0)** | 55.82 **(0.01)**
 :green_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.94 **(0.23)** | 41.67 **(0.84)** | 36.21 **(0)** | 56.99 **(0.25)**</details>

### CODE REVIEW

**Comment by mohanarpit** [comment]
@wmdev0808 This happens in the datasource pane as well. I was trying to use space in the Header fields for an API datasource and faced the same issue. Can you please confirm if this fix will solve the issue for the datasource pane components as well?

**Comment by wmdev0808** [comment]
> @wmdev0808 This happens in the datasource pane as well. I was trying to use space in the Header fields for an API datasource and faced the same issue. Can you please confirm if this fix will solve the issue for the datasource pane components as well?

Hi, I could not reproduce the issue, could you please provide me with a loom video to show the issue?

File: app/client/src/components/ads/DraggableListCard.tsx [hunk 1]
```diff
@@ -116,6 +116,7 @@ export function DraggableListCard(props: RenderComponentProps) {
         onFocus={onFocus}
         placeholder={placeholder}
         ref={ref}
+        trimValue={false}
```
Comment (somangshu) [inline app/client/src/components/ads/DraggableListCard.tsx:119]: @marks0351 @riodeuno are you aware why this was done, I am not sure about the change!

**Review by somangshu — COMMENTED** [review]
Adding other reviews, Not very sure here

File: app/client/src/components/ads/DraggableListCard.tsx (hunk 1 above)
Comment (marks0351) [reply app/client/src/components/ads/DraggableListCard.tsx:119]: I would suggest making changes to the core component, default behaviour should not trim spaces.

Comment (abhvsn) [reply app/client/src/components/ads/DraggableListCard.tsx:119]: IIUC this should happen even if we don’t connect the app with git, is this correct? Also, we haven’t touched the widget level DSLs for git-sync so I’m little skeptical if this is anyway related.

Comment (marks0351) [reply app/client/src/components/ads/DraggableListCard.tsx:119]: I didnt mean the git project caused these changes sry if that sounded so...there were a few changes introduced to text input as part of the project which lead to this behaviour. This has no relation to the git sync feature

Comment (somangshu) [reply app/client/src/components/ads/DraggableListCard.tsx:119]: @marks0351 would it be possible for you to point to the code change / commit?

Comment (marks0351) [reply app/client/src/components/ads/DraggableListCard.tsx:119]: https://github.com/appsmithorg/appsmith/commit/d3c3707ad3fa123bb07119c6c224867883faaa31

Comment (marks0351) [reply app/client/src/components/ads/DraggableListCard.tsx:119]: const { trimValue = true } = props;
should be 
const { trimValue = false } = props;

Comment (marks0351) [reply app/client/src/components/ads/DraggableListCard.tsx:119]: https://github.com/appsmithorg/appsmith/blob/release/app/client/src/components/ads/TextInput.tsx#L303

**Review by Rishabh-Rathod — APPROVED** [review]
LGTM

**Comment by YogeshJayaseelan** [comment]
@wmdev0808 Tested this PR and spaces are now allowed on the widgets (Tab, Table, Menu Button, Button and Button group widgets)

CC: @shwetha-ramesh

**Pull request merged**